checkout-ui-custom/src/_js/_customAddressForm.js

```diff
@@ -165,377 +281,622 @@ class fnsCustomAddressForm {
+              <p class="input ship-number text ${
+                _this.addressrules.number ? 'required' : 'hide'
+              }"><label for="ship-complement">${
+      _this.locale.number ? _this.locale.number : 'Number'
+    }</label><input ${
+      _this.addressrules.number ? 'required' : ''
+    } autocomplete="on" id="ship-number" type="text" name="v-custom-number" maxlength="20" placeholder="${
+      _this.locale.number ? _this.locale.number : ''
+    }" class="input-xlarge" data-hj-whitelist="true" value="${
+      shippingData.address
+        ? shippingData.address.number === null
+          ? ''
+          : shippingData.address.number
+        : ''
+    }"><span class="help error" style="">${
+      _this.locale ? _this.locale.requiredField : 'This field is required.'
+    }</span></p>
+              <p class="input ship-complement text"><label for="ship-complement">${
+                _this.locale
+                  ? _this.locale.address2Placeholder
+                  : 'Apartment number, unit, floor, etc.'
+              }</label><input autocomplete="on" id="ship-complement" type="text" name="v-custom-complement" maxlength="750" placeholder="${
+      _this.locale.address2Placeholder ? _this.locale.address2Placeholder : ''
+    }" class="input-xlarge" data-hj-whitelist="true" value="${
+      shippingData.address
+        ? shippingData.address.complement === null
+          ? ''
+          : shippingData.address.complement
+        : ''
+    }"></p>
             </div>
             <div class="vcustom--vtex-omnishipping-1-x-address__state">
-              <p class="input ship-country text ${_this.deliveryCountries.length<=1 ? "hide" : ""} "><label for="ship-country">Country</label><select name="v-custom-country" id="ship-country" class="input-large">${_this.getCountries().join("")}</select></p>
-              <p class="input ship-city required text"><label for="ship-city">${_this.locale ? _this.locale.city : "City"}</label><input required autocomplete="on" id="ship-city" type="text" name="v-custom-city" maxlength="100" class="input-large" data-hj-whitelist="true" value="${shippingData.address ? shippingData.address.city : "" }"><span class="help error" style="">${_this.locale.requiredField ? _this.locale.requiredField : "This field is required."}</span></p>
-              <p class="input ship-state required text"><label for="ship-state">${_this.locale ? _this.locale.state : "State"}</label>
+              <p class="input ship-country text ${
+                _this.deliveryCountries.length <= 1 ? 'hide' : ''
+              } "><label for="ship-country">Country</label><select name="v-custom-country" id="ship-country" class="input-large">${_this
+      .getCountries()
```

Comment:
As mentioned previously! Seems to be broken `_customAddressForm.js:530 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'join')`